src/blocks/Block.as

```diff
@@ -833,7 +780,13 @@ public class Block extends Sprite {
 		}
 		// TODO: Remove any waiting reporter data in the Scratch.app.extensionManager
 		if (parent is Block) Block(parent).removeBlock(this);
-		else if (parent) parent.removeChild(this);
+		else if (prevBlock) prevBlock.removeBlock(this);
+		else if (parent) {
```

Comment:
I think the `if (parent)` and `if (prevBlock)` checks here might need to be reversed